Composer/packages/client/src/store/reducer/index.ts

```diff
@@ -435,12 +446,39 @@ const publishFailure: ReducerFunc = (state, { error }) => {
 };
 
 const getPublishStatus: ReducerFunc = (state, payload) => {
-  if (payload.results?.botStatus === 'connected') {
+  // the action below only applies to when a bot is being started using the "start bot" button
+  // a check should be added to this that ensures this ONLY applies to the "default" profile.
+  if (payload.target.name === 'default' && payload.endpointURL) {
     state.botStatus = BotStatus.connected;
+    state.botEndpoints[state.projectId] = `${payload.endpointURL || 'http://localhost:3979'}/api/messages`;
+  }
+
+  // if no history exists, create one with the latest status
+  // otherwise, replace the latest publish history with this one
+  if (!state.publishHistory[payload.target.name] && payload.status !== 404) {
+    state.publishHistory[payload.target.name] = [payload];
+  } else if (payload.status !== 404) {
+    // TODO: this should only happen if they actually represent the same item...
```

Comment:
I don't understand the todo comment.

Reply:
I think use status as a part of history when 
1. history not exist, but the profile has lastPublishTime field. It means this bot has published in this profile before, but the bot still publishing and not response. So it call getStatus and use the status as history.
2. history exist, when bot publish to profile, then get 202 response and save in history, which means bot still publishing.  After 10 seconds, Frontend will call getStatus api to get the publish response(success or fail) to update the history. So, maybe the todo comment is this situation.

Reply:
Ben wrote a doc about those api before. https://microsoft-my.sharepoint-df.com/:w:/r/personal/benbro_microsoft_com1/_layouts/15/guestaccess.aspx?e=UadX7x&share=EQUB-_H36KxOu68O_-VujGoBAze7u-3TiyNx38sBogrnww

Reply:
the point of the TODO comment is, there should be some validation that the returned status item actually represents the  0th item in the history array, which is currently being replaced.

there is a scenario where, for example, someone else started a publish just seconds after yours, and the status that gets returned does not match and your publish would just vanish from the list.

I can address this...

Reply:
addressed.